### PR TITLE
infering which loss function to report

### PR DESCRIPTION
**Pull request by mahmoudyusof**
### before
when creating a trainer object, if you use any loss function other than `CrossEntropy` and not specify it in the `metrics` param, you'll get unexpected results

### after 
specifying the loss function will automatically make the trainer object report the correct loss function without specifying the `metrics` param. the `metrics` param can override it, though.

### CODE REVIEW

**Comment by mahmoudyusof** [comment]
@googlebot I signed it!
